src/player.cpp

```diff
@@ -3862,14 +3862,28 @@ void player::check_needs_extremes()
             hp_cur[hp_torso] = 0;
         } else {
             if( calendar::once_every( 1_hours ) ) {
-                if( get_kcal_percent() < 0.1f ) {
-                    add_msg_if_player( m_warning, _( "Food..." ) );
-                } else if( get_kcal_percent() < 0.25f ) {
-                    add_msg_if_player( m_warning, _( "You are STARVING!" ) );
-                } else if( get_kcal_percent() < 0.5f ) {
-                    add_msg_if_player( m_warning, _( "You feel like you haven't eaten in days..." ) );
-                } else if( get_kcal_percent() < 0.8f ) {
-                    add_msg_if_player( m_warning, _( "Your stomach feels so empty..." ) );
+                if( stomach.contains() == 0_ml ) {
+                    if( get_kcal_percent() < 0.1f ) {
+                        add_msg_if_player( m_warning, _( "Food..." ) );
+                    } else if( get_kcal_percent() < 0.25f ) {
+                        add_msg_if_player( m_warning, _( "You are STARVING!" ) );
+                    } else if( get_kcal_percent() < 0.5f ) {
+                        add_msg_if_player( m_warning, _( "You feel like you haven't eaten in days..." ) );
```

Comment:
Still not necessarily true: the player may have just eaten a lettuce an hour ago. Perhaps `"Your stomach feels so empty..."` on this line and `"You feel like you haven't eaten enough..."` on the next line?

Reply:
> the player may have just eaten a lettuce an hour ago

This message will be shown only on empty stomach.
If your stomach is actually empty after eating lettuce an hour ago, I'd say you could say (exaggerating, of course) that you haven't eaten in days.

Reply:
How much calories does a proper (or near-proper meal) restore? If it's less than or around 50% then it's possible for the player to have a meal and then become empty-stomached and still having less than 50% calories. In that case they probably can't say they haven't eaten for days, even for exaggeration.

Reply:
50% of what?
If you mean `get_kcal_percent() < 0.5f` then it's 50% of `healthy_calories`, i.e. 50% of 55000, i.e. 27500 calories. I doubt anyone can eat food that much calories worth in one sitting. We're still talking about one-quarter-full stomach, right?

Reply:
> I doubt anyone can eat food that much calories worth in one sitting.

That's what I mean, even if `get_kcal_percent() < 0.5f` there's still the possibility that the player has eaten a while ago and becomes empty again. In that case they can't say they haven't eaten in days.

Reply:
I'm still not sure. This way we could add `Your stomach feels so empty...` line on every case (80%, 50%, and 10%).